MQ2FeedMe.cpp

```diff
@@ -126,214 +312,370 @@ void ListTypes(const std::list<std::string>& list)
+		WriteChatf("\ay%s\aw:: AutoFeed (\ag%s\ax).", PLUGIN_NAME, (iFeedAt) ? Buffer : "\aroff");
+
+		if (bAnnLevels)
+		{
+			WriteChatf("\ay%s\aw:: Current Thirst (\ag%d\ax) Hunger (\ag%d\ax)", PLUGIN_NAME, GetPcProfile()->thirstlevel, GetPcProfile()->hungerlevel);
+		}
+	}
+
+	GenericCommand(szLine);
+}
+
+void AddDrinkItem()
+{
+	if (!ItemOnCursor())
+	{
+		WriteChatf("%s:: \arNeed to have a food item on your cursor to do this.", PLUGIN_NAME);
+		return;
 	}
-	else if (!_stricmp(Arg, "add")) {
-		if (!ItemOnCursor()) {
-			WriteChatf("%s:: \arNeed to have a food item on your cursor to do this.", PLUGIN_NAME);
+
+	ItemPtr pItem = GetPcProfile()->GetInventorySlot(InvSlot_Cursor);
+	if (pItem)
+	{
+		if (!pItem->GetItemDefinition()->FoodDuration || pItem->GetItemClass() != ItemClass_Drink)
+		{
+			WriteChatf("%s:: \arThat's not a drink. Don't be ridiculous", PLUGIN_NAME);
 			return;
 		}
 
-		ItemPtr pItem = GetPcProfile()->GetInventorySlot(InvSlot_Cursor);
-		if (pItem) {
-			if (!pItem->GetItemDefinition()->FoodDuration || pItem->GetItemClass() != ItemClass_Drink) {
-				WriteChatf("%s:: \arThat's not a drink. Don't be ridiculous", PLUGIN_NAME);
+		WriteChatf("%s:: \ayFound Item: \ap%s", PLUGIN_NAME, pItem->GetName());
+		int FoodIndex = static_cast<int>(Thirst.size()) + 1;
+
+		for (const std::string& itemName : Thirst)
+		{
+			if (ci_equals(itemName, pItem->GetName()))
+			{
+				WriteChatf("%s:: \ap%s \aris already on the list", PLUGIN_NAME, pItem->GetName());
 				return;
 			}
-			WriteChatf("%s:: \ayFound Item: \ap%s", PLUGIN_NAME, pItem->GetName());
-			int FoodIndex = (int)Thirst.size() + 1;
-			for (const std::string& itemName : Thirst) {
-				if (ci_equals(itemName, pItem->GetName())) {
-					WriteChatf("%s:: \ap%s \aris already on the list", PLUGIN_NAME, pItem->GetName());
-					return;
-				}
-			}
-			WritePrivateProfileString("Drink", fmt::format("Drink{}", FoodIndex), pItem->GetName(), INIFileName);
-			Thirst.push_back(pItem->GetName());
-			EzCommand("/autoinv");
-			WriteChatf("%s \agAdded\aw: \ap%s \ayto your autodrink list", PLUGIN_NAME, pItem->GetName());
 		}
+		WritePrivateProfileString("Drink", fmt::format("Drink{}", FoodIndex), pItem->GetName(), INIFileName);
+		Thirst.push_back(pItem->GetName());
+		DoCommand("/autoinv");
+		WriteChatf("%s \agAdded\aw: \ap%s \ayto your autodrink list", PLUGIN_NAME, pItem->GetName());
 	}
-	else if (IsNumber(Arg)) {
-		DrnkAt = atoi(Arg);
-		if (DrnkAt < 0) DrnkAt = 0;
-		else if (DrnkAt > 5000) DrnkAt = 5000;
-		sprintf_s(Buffer, "%d", DrnkAt);
-		WritePrivateProfileString(GetCharInfo()->Name, "AutoDrink", Buffer, INIFileName);
-		WriteChatf("\ay%s\aw:: AutoDrink(\ag%s\ax).", PLUGIN_NAME, (DrnkAt) ? Buffer :"\aroff");
-		if (bAnnLevels) WriteChatf("\ay%s\aw:: Current Thirst(\ag%d\ax) Hunger(\ag%d\ax)", PLUGIN_NAME, GetPcProfile()->thirstlevel, GetPcProfile()->hungerlevel);
+}
+
+void AutoDrinkCmd(PlayerClient* pLPlayer, const char* szLine)
```

Comment:
Does it seem like the methodology for AutoDrink and AutoFeed are going to be quite identical save for the item and checks that you're doing? If it is, you might want to go one more layer of abstraction here rather than having both do the same thing with similar verbosity.

Of course, if the plan is that they diverge that's not necessarily the case.